### DISCUSSION TITLE
feat: add SAP Notes notes download?

### DISCUSSION THREAD

**Discussion by ansibleforsap** [sap-linuxlab/community.sap_launchpad#69] [Ideas]
Hi, would it be possible to add the ability to download SNOTES from the Support Portal? I have an ansible playbook that needs to get the latest compatible OS-release for the respective HANA version (Note 2235581)

**Comment by sean-freeman** [comment]
_NOTE: I would recommend you change your handle...._

@ansibleforsap Yes it would be possible, but the contribution will have to come from the community. See discussion in #15 

To summarise the other issue:
- due to conflict of interest, I would be unable to review any PR for any community contribution
- an Ansible Module would need to be written that re-uses the underlying Python Modules
- the Ansible Module should only retrieve the data from the APIs, and not execute anything on the target system (that can be done by higher-level Ansible Role)
- the API Metadata endpoints will need to interpreted to create this Ansible Module, I have adjusted the below to show which API Metadata endpoints are related to this use case

```
https://api.sap.com/odata/1.0/catalog.svc/$metadata

https://launchpad.support.sap.com/services/odata/svt/snogwslpsearch/$metadata

-- snogwslpsearch/SAPNotes(0000000)
-- snogwslpsearch/ComponentFilterSet('HAN%2A')

https://launchpad.support.sap.com/services/odata/svt/snogwsnnf/$metadata

-- snogwsnnf/Header(Number=0000000,LanguageKey=\'undefined\')
-- snogwsnnf/Doc(Number=0000000,LanguageKey=\'E\')?
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/CVSS
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/SoftComponents
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/SupportPackages
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/Patches
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/CorrInstructions
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/Preconditions
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/ReferencesTo
-- snogwsnnf/Doc(Number=0000000l,LanguageKey=\'E\')/ReferencedBy
```

**Comment by ansibleforsap** [comment]
Hmm i see. I am afraid i lack the neccessary skills to implement what you described :/

What's up with my handle? O.o

**Comment by sean-freeman** [comment]
We can leave this open, I've added a label to do the GH Issue.

Your handle implies you are part of the initiative team, and now you have opened a GH Issue within the GH Org - your handle will appear when people raise new Issues etc.  Changing the handle would avoid that potential outcome, which would potentially spam your email address and also might cause us to miss an end-users help request / comment.
